manifests/init.pp

```diff
@@ -187,106 +192,107 @@
-  Boolean $ldap_log                                                = $rabbitmq::params::ldap_log,
-  Hash $ldap_config_variables                                      = $rabbitmq::params::ldap_config_variables,
-  Integer[1, 65535] $stomp_port                                    = $rabbitmq::params::stomp_port,
-  Boolean $stomp_ssl_only                                          = $rabbitmq::params::stomp_ssl_only,
-  Boolean $wipe_db_on_cookie_change                                = $rabbitmq::params::wipe_db_on_cookie_change,
-  String $cluster_partition_handling                               = $rabbitmq::params::cluster_partition_handling,
-  Variant[Integer[-1,], Enum['unlimited', 'infinity']] $file_limit = $rabbitmq::params::file_limit,
-  Hash $environment_variables                                      = $rabbitmq::params::environment_variables,
-  Hash $config_variables                                           = $rabbitmq::params::config_variables,
-  Hash $config_kernel_variables                                    = $rabbitmq::params::config_kernel_variables,
-  Hash $config_management_variables                                = $rabbitmq::params::config_management_variables,
-  Hash $config_additional_variables                                = $rabbitmq::params::config_additional_variables,
-  Optional[Array] $auth_backends                                   = undef,
-  Optional[String] $key_content                                    = undef,
-  Optional[Integer] $collect_statistics_interval                   = undef,
-  Boolean $ipv6                                                    = $rabbitmq::params::ipv6,
-  String $inetrc_config                                            = $rabbitmq::params::inetrc_config,
-  Stdlib::Absolutepath $inetrc_config_path                         = $rabbitmq::params::inetrc_config_path,
-  Boolean $ssl_erl_dist                                            = $rabbitmq::params::ssl_erl_dist,
-  Optional[String] $rabbitmqadmin_package                          = $rabbitmq::params::rabbitmqadmin_package,
-  Array $archive_options                                           = $rabbitmq::params::archive_options,
-  Array $loopback_users                                            = $rabbitmq::params::loopback_users,
+class rabbitmq (
+  Boolean $admin_enable                                             = $rabbitmq::params::admin_enable,
+  Enum['ram', 'disk', 'disc'] $cluster_node_type                    = $rabbitmq::params::cluster_node_type,
+  Array $cluster_nodes                                              = $rabbitmq::params::cluster_nodes,
+  String $config                                                    = $rabbitmq::params::config,
+  Boolean $config_cluster                                           = $rabbitmq::params::config_cluster,
+  Hash $cluster                                                     = $rabbitmq::params::cluster,
```

Comment:
can you also define the structure within the Hash please?

Reply:
And If possible, don’t change the formatting before that open paren - I think the param spacing also doesn’t need to change, which would make this easier to review?

Reply:
@bastelfreak I don't see a hash in the module that has a definition of it's inner structure should I nevertheless add one?